### PR TITLE
Avoiding trying to load the native library Microsoft.DiaSymReader.Native.<arch>.dll on Linux

### PR DESCRIPTION
**Pull request by neyromant**
Hi, this is attempt to fix problem #92278

### CODE REVIEW

**Review by teo-tsirpanis — COMMENTED** [review]
How about you bail out at the beginning of the method?

**Comment by jkotas** [comment]
All code related to `Microsoft.DiaSymReader` should be under `FEATURE_ISYM_READER`. Looks like there is a bug in that `FEATURE_ISYM_READER` is defined on all platforms. It should be defined on Windows only.

Could you please try to fix this by defining `FEATURE_ISYM_READER` on Windows only here: https://github.com/dotnet/runtime/blob/2000251605d7978979d1fef101673335513bd765/src/coreclr/clrdefinitions.cmake#L137

**Comment by neyromant** [comment]
@jkotas thank you. I added a condition for adding a definition only for windows

**Review by jkotas — APPROVED** [review]
LGTM! Thank you

File: src/coreclr/clrdefinitions.cmake [hunk 1]
```diff
@@ -134,7 +134,9 @@ endif (CLR_CMAKE_TARGET_WIN32 AND (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARG
 if(FEATURE_INTERPRETER)
   add_compile_definitions(FEATURE_INTERPRETER)
```
Comment (jkotas) [inline src/coreclr/clrdefinitions.cmake:135]:
```suggestion
    add_compile_definitions(FEATURE_INTERPRETER)
```
Fix indentation

**Comment by neyromant** [comment]
> LGTM! Thank you

@jkotas Thank you! 
But after merge main branch for resolve conflicts some tests, even those that previously completed successfully, failed.
Tell me, please, what to do?

**Comment by jkotas** [comment]
The failure is not related to this PR. I have opened https://github.com/dotnet/runtime/issues/92518 on it.

**Review by tommcdon — APPROVED** [review]
Looks good to me.  Thanks for the contribution!

**Pull request merged**